umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso19115_2/data_contact.clj

```diff
@@ -67,7 +67,8 @@
  [contact-info-xml url-content-type sanitize?]
  (for [url (select contact-info-xml "gmd:onlineResource/gmd:CI_OnlineResource")
        :let [name (char-string-value url "gmd:name")
-             url-link (value-of url "gmd:linkage/gmd:URL")]]
+             url-link (value-of url "gmd:linkage/gmd:URL")]
+       :when url-link]
```

Comment:
The (when url-link (url/format-url url-link sanitize?)) code just formats the url and sanitizes it if the flag is true when the url-link exists, otherwise just the url-link is nil but the rest of the object is filled out and returned.  We don't want that. If the url link doesn't exist then we shouldn't create a related url object. I do agree though that I can get rid of the (when url-link code below and just replace it with :URL (url/format-url url-link sanitize?) since by this point then (when url-link will always be true.

Reply:
I have fixed this issue by removing the (when url-link just for the :URL.